dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -1309,12 +1376,19 @@ CFileDefExt::CountFolderAndFiles(HWND hwndDlg, LPWSTR pwszBuf, UINT cchBufMax, D
     /* Find filename position */
     UINT cchBuf = wcslen(pwszBuf);
     WCHAR *pwszFilename = pwszBuf + cchBuf;
+    WCHAR PathBuffer[MAX_PATH] = L"";
+    WCHAR WorkBuffer[MAX_PATH] = L"";
```

Comment:
Any reason for the fixed size buffers?

Reply:
Consistency with other path-related buffers in that module, which are all fixed sized to MAX_PATH

Reply:
Which is a bug that should be fixed, that something is done wrong does not mean that it should be continued that way

Reply:
Why is using a fixed size buffer of MAX_PATH a bug when sole purpose is to store a file path ?
Grepping [MAX_PATH] shows a few occurence in ROS overall.... What's different here ?

Reply:
Because this is not the maximum path size that a path can be, and ROS (especially explorer) should not limit that to this size.

Reply:
OK, i'll play with HeapAlloc/Free. Btw, this allowed me to point out a memory leak in BOOL CFileVersionInfo::Load(LPCWSTR pwszPath) as m_pInfo is never free'd. I'll fix this as well.